File: Java(eclipse)/workspace_ServletAndJSP/書籍検索アプリ/src/main/java/todo/model/CsvData.java
package todo.model;

import todo.model.beans.BookDataBean;

public class CsvData implements ILoadData{
	
	private CsvReader csvReader;
    private BookDataBean[] bookDataBeans;
    
    //---------
    //コンストラクタ
    public CsvData() {
    	
    	bookDataBeans = new BookDataBean[0];
		csvReader = new CsvReader(bookDataBeans);
    }
    public CsvData(BookDataBean[] bookDataBeans) {
		
		this.bookDataBeans = bookDataBeans;
		csvReader = new CsvReader(bookDataBeans);
	}
	
	public BookDataBean[] loadBookDatas(String filePath) {
		
        return bookDataBeans = csvReader.readCsv(filePath);
    }
	
	public BookDataBean[] getCloneAllBookData() {
		
		return bookDataBeans.clone();
	}
	
	public BookDataBean[] getBookDataBeans() {
		
		return bookDataBeans;
	}
	
	@Override
	public BookDataBean[] BookDataBeanGet() {
		
		return getBookDataBeans();
	}
}

File: 1_C#(VisualStudio)/1_workspace_ASP.NETCoreMVC/書籍検索アプリ_移植元Java/Models/CsvData.cs
using 書籍検索アプリ_移植元Java.Models.beans;
using 書籍検索アプリ_移植元Java.Models.interfaces;

namespace 書籍検索アプリ_移植元Java.Models {
    public class CsvData : ILoadData {

        private readonly CSVReader csvReader;
        private BookDataBean[] bookDataBeans;

        //---------
        //コンストラクタ
        public CsvData() {

            bookDataBeans = Array.Empty<BookDataBean>();
            csvReader = new CSVReader(bookDataBeans);
        }
        public CsvData(BookDataBean[] bookDataBeans) {

            this.bookDataBeans = bookDataBeans;
            csvReader = new CSVReader(bookDataBeans);
        }

        public BookDataBean[] LoadBookDatas(string filePath) {

            return bookDataBeans = csvReader.ReadCsv(filePath);
        }

        public BookDataBean[] GetCloneAllBookData() {

            return (BookDataBean[])bookDataBeans.Clone();
        }

        public BookDataBean[] GetBookDataBeans() {

            return bookDataBeans;
        }

        public BookDataBean[] BookDataBeanGet() {

            return GetBookDataBeans();
        }
    }
}